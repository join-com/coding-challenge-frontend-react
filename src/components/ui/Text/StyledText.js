import styled from 'styled-components';
import {fonts, fontSizes, colors} from '../../../constants/styles'


const StyledText = styled.p`
  position: relative;
  font-family: ${fonts.primary};
  font-size: ${({ size = 'normal' }) => fontSizes[size]}rem;
  font-weight: ${({ bold }) => (bold ? 'bold' : 500)};
  font-style: normal;
  font-stretch: normal;
  line-height: 1.2;
  letter-spacing: normal;
  text-align: ${({ textAlign = 'left' }) => textAlign};
  color: ${({ inverted }) => (inverted ? colors.light : colors.dark)};
  text-transform: ${({ uppercase }) => (uppercase ? 'uppercase' : 'none')};
  margin: ${({ size = 'normal' }) => `0 0 ${fontSizes[size] / 2}rem 0`};
`;

export const StyledErrorText = styled(StyledText)`
  color: ${colors.error};
`;

export default StyledText;
