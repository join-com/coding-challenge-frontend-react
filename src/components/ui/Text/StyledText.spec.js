import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import StyledText, { StyledErrorText } from "./StyledText";

describe("StyledText component", () => {
  const defaultProps = {
    disabled: false
  };

  const getWrapper = (testProps = {}) => {
    const props = { ...defaultProps, ...testProps };
    return renderer.create(<StyledText {...props} />).toJSON();
  };

  it("should match snapshot ", () => {
    const wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot for different size prop ", () => {
    const wrapper = getWrapper({ size: "large" });
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot for uppercase prop ", () => {
    const wrapper = getWrapper({ uppercase: true });
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot for textAlign prop ", () => {
    const wrapper = getWrapper({ textAlign: "center" });
    expect(wrapper).toMatchSnapshot();
  });
});

describe("StyledErrorText component", () => {
  it("should match snapshot ", () => {
    expect(renderer.create(<StyledErrorText />).toJSON()).toMatchSnapshot();
  });
});
