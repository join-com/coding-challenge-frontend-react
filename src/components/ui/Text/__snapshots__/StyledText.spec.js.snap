// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledErrorText component should match snapshot  1`] = `
.c0 {
  position: relative;
  font-family: 'Oswald',sans-serif;
  font-size: 1rem;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.2;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-align: left;
  color: #1D232B;
  text-transform: none;
  margin: 0 0 0.5rem 0;
  color: #DC3545;
}

<p
  className="c0"
/>
`;

exports[`StyledText component should match snapshot  1`] = `
.c0 {
  position: relative;
  font-family: 'Oswald',sans-serif;
  font-size: 1rem;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.2;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-align: left;
  color: #1D232B;
  text-transform: none;
  margin: 0 0 0.5rem 0;
}

<p
  className="c0"
  disabled={false}
/>
`;

exports[`StyledText component should match snapshot for different size prop  1`] = `
.c0 {
  position: relative;
  font-family: 'Oswald',sans-serif;
  font-size: 1.5rem;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.2;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-align: left;
  color: #1D232B;
  text-transform: none;
  margin: 0 0 0.75rem 0;
}

<p
  className="c0"
  disabled={false}
  size="large"
/>
`;

exports[`StyledText component should match snapshot for textAlign prop  1`] = `
.c0 {
  position: relative;
  font-family: 'Oswald',sans-serif;
  font-size: 1rem;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.2;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-align: center;
  color: #1D232B;
  text-transform: none;
  margin: 0 0 0.5rem 0;
}

<p
  className="c0"
  disabled={false}
/>
`;

exports[`StyledText component should match snapshot for uppercase prop  1`] = `
.c0 {
  position: relative;
  font-family: 'Oswald',sans-serif;
  font-size: 1rem;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.2;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-align: left;
  color: #1D232B;
  text-transform: uppercase;
  margin: 0 0 0.5rem 0;
}

<p
  className="c0"
  disabled={false}
/>
`;
